discovery-provider/src/tasks/index_metrics.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from urllib.parse import urljoin
+import requests
+import dateutil.parser
+from sqlalchemy import func, desc, or_, and_
+from src.models import RouteMetrics, AppNameMetrics
+from src.tasks.celery_app import celery
+
+from src.utils.redis_metrics import metrics_prefix, metrics_application, \
+    metrics_routes, parse_metrics_key, get_rounded_date_time
+
+logger = logging.getLogger(__name__)
+
+def process_route_keys(session, redis, key, date):
+    route_metrics = []
+    routes = redis.hgetall(key)
+    for key_bstr in routes:
+        route = key_bstr.decode('utf-8').strip('/')
+        val = int(routes[key_bstr].decode('utf-8'))
+
+        version = None
+        path = route
+        query_string = None
+
+        route_subpaths = route.split('/') 
+
+        # Extract the version out of the path
+        if route_subpaths[0].startswith('v') and len(route_subpaths[0]) > 1:
+            version = route_subpaths[0][1:]
+            path = '/'.join(route_subpaths[1:])
+        
+        # Extract the query string out of the path
+        route_query = path.split('?') 
+        if len(route_query) > 1:
+            path = route_query[0]
+            query_string = route_query[1]
+        route_metrics.extend([
+            RouteMetrics(
+                version = version,
+                route_path = path,
+                query_string = query_string,
+                count = val,
+                timestamp = date
+            )
+        ])
+
+    if route_metrics:
+        session.bulk_save_objects(route_metrics)
+    redis.delete(key)
+
+    return route_metrics
+
+def process_app_name_keys(session, redis, key, date):
+    app_name_metrics = []
+    routes = redis.hgetall(key)
+    for key_bstr in routes:
+        app_name = key_bstr.decode('utf-8')
+        val = int(routes[key_bstr].decode('utf-8'))
+
+        app_name_metrics.extend([
+            AppNameMetrics(
+                application_name = app_name,
+                count = val,
+                timestamp = date
+            )
+        ])
+    if app_name_metrics:
+        session.bulk_save_objects(app_name_metrics)
+    redis.delete(key)
+
+
+def update_route_metrics(db, redis):
+    with db.scoped_session() as session:
+        all_keys = redis.keys(f"{metrics_prefix}:*")
```

Comment:
doing keys * is generally not best practice https://redis.io/commands/keys. in this case it's probably okay cause we dont have a ton of keys in dp like cn or identity. the suggested way is probably SCAN, but we don't need to change it at the moment

Reply:
okay i actually checked and we have like ~20 keys in redis in dp. we're totally fine

Reply:
I just updated it to use scan.